eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -105,9 +105,9 @@ object OnionMessages {
           case Left(_) => None
           case Right(decoded) =>
             decoded.tlvs.get[RouteBlindingEncryptedDataTlv.OutgoingNodeId] match {
-              case None => None
-              case Some(RouteBlindingEncryptedDataTlv.OutgoingNodeId(nextNodeId)) =>
+              case Some(RouteBlindingEncryptedDataTlv.OutgoingNodeId(EncodedNodeId.Plain(nextNodeId))) =>
                 Some(Sphinx.RouteBlinding.BlindedRoute(nextNodeId, decoded.nextBlinding, route.blindedNodes.tail))
+              case _ => None // TODO: allow compact node id
```

Comment:
It's a bit unclear what we want to do here, and it seems to me that we currently don't support a similar scenario added by https://github.com/lightning/bolts/commit/ba2b95a98a415a24c27685a34648c1f64468a4bf

Since it looks like we'll need the ability to resolve an scid to a `node_id`, this will impact the whole design of this component, this cannot be a plain function anymore unfortunately...but that re-work should be done in a follow-up PR, once we have more clarity on https://github.com/lightning/bolts/pull/1138

Reply:
Actually I'm a bit lost here, because it looks like https://github.com/ACINQ/eclair/pull/2762 added support for relaying using `channel_id`, but did we miss that specific edge case?

Reply:
You're right, #2762 missed the edge case of when we are the introduction node. This is annoying, there is no easy way to fix it. I'll try to figure out the best way to do it.